### PR TITLE
New XSC analytics metrics capabilities

### PR DESCRIPTION
**Pull request by gailazar300**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

1. Add XscAuthConfig.
2. Add XSCmock server.
3. Add technology to language mapping.

depends on: https://github.com/jfrog/jfrog-client-go/pull/928

### CODE REVIEW

File: utils/config/config.go [hunk 1]
```diff
@@ -708,6 +709,19 @@ func (serverDetails *ServerDetails) CreateXrayAuthConfig() (auth.ServiceDetails,
 	return serverDetails.createAuthConfig(artAuth)
 }
 
+func (serverDetails *ServerDetails) CreateXscAuthConfig() (auth.ServiceDetails, error) {
+	ascAuth := xscAuth.NewXscDetails()
+	ascAuth.SetUrl(convertXrayUrlToXscUrl(serverDetails.XrayUrl))
```
Comment (attiasas) [inline utils/config/config.go:714]:
```suggestion
	ascAuth.SetUrl(serverDetails.getXscUrlFromXrayUrl())
```
private method, called once, passing an argument that is also an attribute.
should use different implementation

File: utils/config/config.go [hunk 2]
```diff
@@ -708,6 +709,19 @@ func (serverDetails *ServerDetails) CreateXrayAuthConfig() (auth.ServiceDetails,
...
+	return serverDetails.createAuthConfig(ascAuth)
+}
+
+// Xray ans Xsc will always have the same platform url.
```
Comment (attiasas) [inline utils/config/config.go:718]:
```suggestion
// Xray and Xsc will always have the same platform url.
```

File: utils/config/config.go [hunk 3]
```diff
@@ -708,6 +709,19 @@ func (serverDetails *ServerDetails) CreateXrayAuthConfig() (auth.ServiceDetails,
...
+func convertXrayUrlToXscUrl(xrayUrl string) string {
+	xscUrl := strings.TrimSuffix(xrayUrl, "/")
+	xscUrl = strings.TrimSuffix(xscUrl, "xray")
+	return xscUrl + "xsc/"
+}
```
Comment (attiasas) [inline utils/config/config.go:723]:
```suggestion
func (serverDetails *ServerDetails) getXscUrlFromXrayUrl() string {
	xscUrl := strings.TrimSuffix(serverDetails.XrayUrl, "/")
	xscUrl = strings.TrimSuffix(xscUrl, "xray")
	return xscUrl + "xsc/"
}
```

File: utils/coreutils/techutils.go [hunk 4]
```diff
@@ -29,13 +29,30 @@ const (
 	Go     Technology = "go"
 	Pip    Technology = "pip"
 	Pipenv Technology = "pipenv"
+	Pypi   Technology = "pypi"
```
Comment (attiasas) [inline utils/coreutils/techutils.go:32]: Pypi is not a technology.
we have 3 tools for python: `Pip`, `Pipenv` and `Poetry`
```suggestion
	Pypi = "pypi"
```

File: utils/coreutils/techutils.go [hunk 5]
```diff
@@ -29,13 +29,30 @@ const (
...
+		Go:     "Go",
+		Maven:  "Java",
+		Gradle: "Java",
+		Nuget:  "C#",
+		Dotnet: "C#",
+		Yarn:   "JavaScript",
+		Pnpm:   "JavaScript",
+	}
+	return languageMap[technology]
+}
```
Comment (attiasas) [inline utils/coreutils/techutils.go:55]:
```suggestion
type CodeLanguage string

const (
    JavaScript CodeLanguage = "javascript"
    Python     CodeLanguage = "python"
    GoLang     CodeLanguage = "go"
    Java       CodeLanguage = "java"
    CSharp     CodeLanguage = "C#"

)
```
and add `Language` attribute to the `TechData` objects

**Review by attiasas — CHANGES_REQUESTED** [review]
Update the PR name to fit the changes better.
Add at the description the changes that were made.

**Review by attiasas — CHANGES_REQUESTED** [review]
Add tests to check connection of language and tech at `techutils_test.go`

File: common/tests/utils.go [hunk 6]
```diff
@@ -76,6 +76,15 @@ func CreateDsRestsMockServer(t *testing.T, testHandler restsTestHandler) (*httpt
...
 }
 
+func CreateXscRestsMockServer(t *testing.T, testHandler restsTestHandler) (*httptest.Server, *config.ServerDetails, artifactory.ArtifactoryServicesManager) {
+	testServer := CreateRestsMockServer(testHandler)
+	serverDetails := &config.ServerDetails{Url: testServer.URL + "/", XrayUrl: testServer.URL + "/xray/"}
+
+	serviceManager, err := utils.CreateServiceManager(serverDetails, -1, 0, false)
+	assert.NoError(t, err)
+	return testServer, serverDetails, serviceManager
+}
```
Comment (attiasas) [inline common/tests/utils.go:86]: This can be created at: https://github.com/jfrog/jfrog-cli-security/pull/47
where it is used. We probably will only use mocks for Xray in our repository
added to `tests/utils/test_utils.go` / `tests/utils/test_config.go`

File: utils/config/config.go [hunk 7]
```diff
@@ -572,6 +573,7 @@ type ServerDetails struct {
 	ArtifactoryUrl                  string `json:"artifactoryUrl,omitempty"`
 	DistributionUrl                 string `json:"distributionUrl,omitempty"`
 	XrayUrl                         string `json:"xrayUrl,omitempty"`
+	XscUrl                          string `json:"xscUrl,omitempty"`
```
Comment (attiasas) [inline utils/config/config.go:576]: Why did you add that if it is never used?
at client-go you are using GetUrl() and you are setings using the value at `XrayUrl`

File: utils/coreutils/techutils.go [hunk 8]
```diff
@@ -37,6 +37,33 @@ const (
...
+		Go:     GoLang,
+		Maven:  Java,
+		Gradle: Java,
+		Nuget:  CSharp,
+		Dotnet: CSharp,
+		Yarn:   JavaScript,
+		Pnpm:   JavaScript,
+	}
+	return languageMap[technology]
+}
```
Comment (attiasas) [inline utils/coreutils/techutils.go:55]: Instead of this func, you can attach the value directly to the `TechData` as new attribute `Language CodeLanguage`

**Review by attiasas — CHANGES_REQUESTED** [review]

File: utils/config/config.go (hunk 7 above)
Comment (gailazar300) [reply utils/config/config.go:576]: we use this field for logic in security-cli

File: utils/coreutils/techutils.go (hunk 8 above)
Comment (gailazar300) [reply utils/coreutils/techutils.go:55]: I don't understand your comment, but what is wrong with this implementation?

File: common/tests/utils.go (hunk 6 above)
Comment (gailazar300) [reply common/tests/utils.go:86]: all other mocks are there, so it makes sense that they will all be together in the same place.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
The code introduced in this PR should be added to the `jfrog-cli-security` project.
